lib/posix/options/pthread.c

```diff
@@ -1046,20 +1046,34 @@ void pthread_exit(void *retval)
 }
 
 /**
- * @brief Wait for a thread termination.
+ * @brief Await a thread termination with timeout.
  *
- * See IEEE 1003.1
+ * Non-portable GNU extension of IEEE 1003.1
  */
-int pthread_join(pthread_t pthread, void **status)
+int pthread_timedjoin_np(pthread_t pthread, void **status, const struct timespec *abstime)
```

Comment:
One possibly cleaner way to go about this is to convert all absolute `struct timespec` to `k_timeout` at the point of entry, and have an internal `pthread_timedjoin_internal(pthread_t th, void **status, k_timeout_t timeout)` (where the timeout is already specified as a `k_timeout_t` instead of `struct timespec`).

That way, e.g.
* `pthread_join(th, st)` -> `pthread_timedjoin_internal(th, st, K_FOREVER)`
* `pthread_tryjoin(th, st)` -> `pthread_timedjoin_internal(th, st, K_NO_WAIT)`
* `pthread_timedjoin_np(th, st, to)` -> `pthread_timedjoin_internal(th, st, K_MSEC(timespec_to_timeoutms(to)))`